lib/core/src/irods_random.cpp

```diff
@@ -1,7 +1,34 @@
+#include "irods/irods_random.hpp"
+
+#include "irods/irods_exception.hpp"
+#include "irods/rodsErrorTable.h"
+
 #include <boost/random.hpp>
+#include <fmt/format.h>
 #include <openssl/rand.h>
-#include "irods/irods_random.hpp"
+
 #include <ctime>
+#include <random>
+#include <string>
+
+namespace
+{
+    auto generate_random_alphanumeric_character() -> char
+    {
+        constexpr char min_char = '0';
+        constexpr char max_char = 'z';
+
+        static std::random_device rd;
+        static std::default_random_engine e{rd()};
+        static std::uniform_int_distribution<> d{min_char, max_char};
+
+        char c{};
+        while (!std::isalnum(c)) {
```

Comment:
Missing cast to `unsigned char`.

See https://en.cppreference.com/w/cpp/string/byte/isalnum.

Reply:
Oh, good to know. Will do